x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/percentile_ranks.tsx

```diff
@@ -0,0 +1,245 @@
+        field: column.sourceField,
+        values: [column.params.value],
+        // time shift is added to wrapping aggFilteredMetric if filter is set
+        timeShift: column.filter ? undefined : column.timeShift,
+      }
+    ).toAst();
+  },
+  getErrorMessage: (layer, columnId, indexPattern) =>
+    combineErrorMessages([
+      getInvalidFieldMessage(
+        layer.columns[columnId] as FieldBasedIndexPatternColumnMultipleValues,
+        indexPattern
+      ),
+      getDisallowedPreviousShiftMessage(layer, columnId),
+    ]),
+  paramEditor: function PercentileParamEditor({
+    layer,
+    updateLayer,
+    currentColumn,
+    columnId,
+    indexPattern,
+  }) {
+    const onChange = useCallback(
+      (value) => {
+        if (!isValidNumber(value) || Number(value) === currentColumn.params.value) {
+          return;
+        }
+        updateLayer({
+          ...layer,
+          columns: {
+            ...layer.columns,
+            [columnId]: {
+              ...currentColumn,
+              label: currentColumn.customLabel
+                ? currentColumn.label
+                : ofName(
+                    indexPattern.getFieldByName(currentColumn.sourceField)?.displayName ||
+                      currentColumn.sourceField,
+                    Number(value),
+                    currentColumn.timeShift
+                  ),
+              params: {
+                ...currentColumn.params,
+                value: Number(value),
+              },
+            } as PercentileRanksIndexPatternColumn,
+          },
+        });
+      },
+      [updateLayer, layer, columnId, currentColumn, indexPattern]
+    );
+    const { inputValue, handleInputChange: handleInputChangeWithoutValidation } = useDebouncedValue<
+      string | undefined
+    >(
+      {
+        onChange,
+        value: String(currentColumn.params.value),
+      },
+      { allowFalsyValue: true }
+    );
+    const inputValueIsValid = isValidNumber(inputValue);
+
+    const handleInputChange: EuiFieldNumberProps['onChange'] = useCallback(
+      (e) => {
+        handleInputChangeWithoutValidation(e.currentTarget.value);
+      },
+      [handleInputChangeWithoutValidation]
+    );
+
+    return (
+      <EuiFormRow
+        label={i18n.translate('xpack.lens.indexPattern.percentile.percentileRanksValue', {
+          defaultMessage: 'Percentile ranks value',
+        })}
+        data-test-subj="lns-indexPattern-percentile_ranks-form"
+        display="rowCompressed"
+        fullWidth
+        isInvalid={!inputValueIsValid}
+        error={
+          !inputValueIsValid &&
+          i18n.translate('xpack.lens.indexPattern.percentileRanks.errorMessage', {
+            defaultMessage: 'Percentile ranks value must be a number',
+          })
+        }
+      >
+        <EuiFieldNumber
+          data-test-subj="lns-indexPattern-percentile_ranks-input"
+          compressed
+          value={inputValue ?? ''}
+          onChange={handleInputChange}
```

Comment:
This has the same issue as percentiles in that it's not possible to order terms by a decimal point percentile value: https://github.com/elastic/elasticsearch/issues/66677

The query fails if you try to do it on this PR. We can either limit it in the same way (enforce integers) or we can drop out of the terms sorting and fall back to alphabetical. What do you think @stratoula @ghudgins ? I'm not sure tbh - integers make more sense for percentile because the value range is strictly 0-100 which is not the case for percentile rank (it can have any data range so you wouldn't be able to use it for fields which have a domain of `0-1` which would be really annoying).

Reply:
Awww nice catch. I was always using integer values and never caught it. Hmmm, I don't know either but I feel that as a user I would prefer adding whatever value I want and not be limited only to integers

Reply:
Check this comment https://github.com/elastic/kibana/pull/132430#issuecomment-1138351115